File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        List<int[]> res= new ArrayList<>();
        
        if(intervals == null || intervals.length == 0 ){
            return res.toArray(new int[0][]);
        }
        Arrays.sort(intervals, (a , b) -> a[0] - b[0]);
        int start = intervals[0][0];
        int end = intervals[0][1];
        
        for(int[] it : intervals){
            if(it[0]<=end){
                end = Math.max(end,it[1]);
            }
            else{
                res.add(new int[]{start,end});
                start = it[0];
                end = it[1];
            }
        }
        res.add(new int[]{start,end});
        return res.toArray(new int[0][]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& interval) {
        vector<vector<int>> mergeInteval;
        if(interval.size() == 0) return mergeInteval;
        sort(interval.begin() , interval.end());
        vector<int> tempInterval= interval[0];
        
        for(auto it : interval){
            if(it[0] <= tempInterval[1]){
                tempInterval[1] = max(it[1] , tempInterval[1]);
            }
            else{
                mergeInteval.push_back(tempInterval);
                tempInterval = it;
            }
        }
        mergeInteval.push_back(tempInterval);
        return mergeInteval;
    }
};